subsys/net/lib/openthread/platform/alarm.c

```diff
@@ -126,3 +126,10 @@ uint32_t otPlatAlarmMicroGetNow(void)
 {
 	return (uint32_t)(k_ticks_to_us_floor64(k_uptime_ticks()) - time_offset_us);
 }
+
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
```

Comment:
Can we implement the function unconditionally? If it's not used, the linker will just drop it.